compiler/rustc_codegen_ssa/src/back/archive.rs

```diff
@@ -231,7 +231,11 @@ impl<'a> ArArchiveBuilder<'a> {
             "gnu" => ArchiveKind::Gnu,
             "bsd" => ArchiveKind::Bsd,
             "darwin" => ArchiveKind::Darwin,
-            "coff" => ArchiveKind::Coff,
+            "coff" => {
+                // FIXME: ar_archive_writer doesn't support COFF archives yet.
+                // https://github.com/rust-lang/ar_archive_writer/issues/9
+                ArchiveKind::Gnu
```

Comment:
Is this path just not hit? If it is hit, does generating the archive in ELF format cause linking issues or something?

Reply:
It is hit, this is used by the cranelift and GCC backends.

GNU != ELF - the names are a bit misleading but can be thought of as GNU == "What GCC would produce for this target" and COFF == "What MSVC would produce for this target". Before I made this change, MSVC targets on the LLVM backend were also using GNU archive formats.